src/python/tests/gff3/test_restructure.py

```diff
@@ -0,0 +1,314 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `ensembl.io.genomio.gff3.restructure` module."""
+
+from contextlib import nullcontext as does_not_raise
+from typing import Any, ContextManager, Dict, List, Union
+
+from Bio.SeqFeature import SeqFeature, SimpleLocation
+import pytest
+from pytest import param, raises
+
+from ensembl.io.genomio.gff3.exceptions import GFFParserError
+from ensembl.io.genomio.gff3.restructure import (
+    restructure_gene,
+    add_transcript_to_naked_gene,
+    move_only_cdss_to_new_mrna,
+    move_only_exons_to_new_mrna,
+    move_cds_to_existing_mrna,
+    remove_extra_exons,
+)
```

Comment:
Just a suggestion for next time: since your method names are not too long, I find it "simpler" for the import stage to just do `from ensembl.io.genomio.gff3 import restructure` and then use `restructure.` when I need something from the class.

Reply:
Right that makes sense, at least it did ensure that I was keeping track of which functions I was testing